sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Regex.java

```diff
@@ -71,7 +138,46 @@ public static Matches matches(String regex, int group) {
    * @param valueGroup The Regex group to use the value
    */
   public static MatchesKV matchesKV(String regex, int keyGroup, int valueGroup) {
-    return new MatchesKV(regex, keyGroup, valueGroup);
+    return matchesKV(Pattern.compile(regex), keyGroup, valueGroup);
+  }
+
+  /**
+   * Returns a {@link Regex.MatchesKV} {@link PTransform} that checks if the entire line matches the
+   * Regex. Returns the specified groups as the key and value as a {@link PCollection}.
+   *
+   * @param pattern The regular expression to run
+   * @param keyGroup The Regex group to use as the key
+   * @param valueGroup The Regex group to use the value
+   */
+  public static MatchesKV matchesKV(Pattern pattern, int keyGroup, int valueGroup) {
+    return new MatchesKV(pattern, keyGroup, valueGroup);
+  }
+
+  /**
+   * Returns a {@link Regex.MatchesNameKV} {@link PTransform} that checks if the
+   * entire line matches the Regex. Returns the specified groups as the key and value as a
+   * {@link PCollection}.
+   *
+   * @param regex The regular expression to run
+   * @param keyGroupName The Regex group name to use as the key
+   * @param valueGroupName The Regex group name to use the value
+   */
+  public static MatchesNameKV matchesKV(String regex, String keyGroupName, String valueGroupName) {
```

Comment:
What about just saying the key group has to be named `key` and the value group has to be named `value`? Do we think that someone is going to have a regex lying around that they re-use for this transform? I bet they are always writing the regex just for this. So we don't really need to parameterize at all and don't need the `int` based version either.